Add initial documentation for the execute_test command (#198)
opensearch-project/opensearch-benchmark: docs/api/execute-test.md

nav_order: 10
---

> The `execute_test` command is the primary usage of OpenSearch Benchmark. This command is used to execute a test.

Reviewer: I'd prefer that, but do not have a strong opinion

## Syntax 

```bash

> opensearch-benchmark execute_test <args>

Reviewer 2: ```suggestion
opensearch-benchmark execute_test <arguments>
```

```

### Arguments

> **Note**: This is not an exhaustive list of arguments. 

Reviewer: Any specific reason we're not able to make it exhaustive list?

Author: There's somewhere around 20 more arguments to document that deal with things like specific configurations for provisioning or reporting. I didn't think it was worth taking the time to add those here since they're used for very advanced use cases. 

I can add them in but it with take a little while

Author: I can add a section to the bottom for "Advanced Arguments" or something that lists them all

Author: These are already documented in the help text 🤦‍♂️ I'll update the table

Author: Decided to split the exhaustive list into a separate table to avoid burying the examples

Argument | Description | Required
:--- | :--- |:---

> `workload` | The dataset and operations that will be executed during a test. Workloads are defined in the [OpenSearch Benchmark Workloads repository](https://github.com/opensearch-project/opensearch-benchmark-workloads). | Yes

Reviewer 2: ```suggestion
`workload` | The dataset and operations that execute during a test. See [OpenSearch Benchmark Workloads repository](https://github.com/opensearch-project/opensearch-benchmark-workloads) for more details on workloads. | Yes
```

> `workload-params` | Parameters defined within each workload that can be overwritten. These are defined within the README of each workload. An example for the eventdata workload can be found [here](https://github.com/opensearch-project/opensearch-benchmark-workloads/tree/main/eventdata#parameters).  | No

Reviewer 2: ```suggestion
`workload-params` | Parameters defined within each workload that can be overwritten, defined in the README of each workload. You can find an example of the eventdata workload [here](https://github.com/opensearch-project/opensearch-benchmark-workloads/tree/main/eventdata#parameters).  | No
```

> `test_procedure` | Test Procedures define the sequence of operations and parameters for a specific workload. If no `test_procedure` is specified the default will be selected. Test Procedures are defined within each workload directory. An example for the eventdata workload can be found [here](https://github.com/opensearch-project/opensearch-benchmark-workloads/blob/main/eventdata/test_procedures/default.json). | No

Reviewer 2: ```suggestion
`test_procedure` | Test Procedures define the sequence of operations and parameters for a specific workload. When there is `test_procedure` specified, Benchmark selects the default. Define test procedures within each workload directory. You can fine an example test procedure [here](https://github.com/opensearch-project/opensearch-benchmark-workloads/blob/main/eventdata/test_procedures/default.json). | No
```

`client-options` | Options for the [OpenSearch Python client](https://opensearch.org/docs/latest/clients/python/). Required if testing against a cluster with security enabled. | No

> `pipeline` | Steps required to execute a test. This can including provisioning an OpenSearch from source code or from a specified distribution. Defaults to `from-sources` which provisions an OpenSearch cluster from source code. | No

Reviewer 2: ```suggestion
`pipeline` | Steps required to execute a test, including provisioning an OpenSearch from source code or a specified distribution. Defaults to `from-sources` which provisions an OpenSearch cluster from source code. | No
```

> `distribution-version` | The OpenSearch version to use for a given test. This can be useful when using a `pipeline` which includes provisioning. When using a `pipeline` without provisioning Benchmark will automatically determine the version | No

Reviewer 2: ```suggestion
`distribution-version` | The OpenSearch version to use for a given test. Defining a version can be useful when using a `pipeline` that includes provisioning. When using a `pipeline` without provisioning, Benchmark will automatically determine the version | No
```

> `target-hosts` | The OpenSearch endpoint(s) to execute a test against. This should only be specified with  `--pipeline=benchmark-only`  | No

Reviewer 2: ```suggestion
`target-hosts` | The OpenSearch endpoint(s) to execute a test against. Should only be specified with  `--pipeline=benchmark-only`  | No
```

> `test-mode` | Run a single iteration of each operation in the Test Procedure. This is meant to provide a quick way for sanity checking a testing configuration and should not be used for actual benchmarking. | No

Reviewer 2: ```suggestion
`test-mode` | Run a single iteration of each operation in the test procedure. The test provides a quick way for sanity checking a testing configuration. Therefore, do not use `test-mode` for actual benchmarking. | No
```

> `kill-running-processes` | Kill any running OpenSearch Benchmark processes on the local machine before the test is executed.  | No

Reviewer 2: ```suggestion
`kill-running-processes` | Kill any running OpenSearch Benchmark processes on the local machine before the test executes. | No
```

nav_order: 10
---

The `execute_test` command of OpenSearch Benchmark executes tests against your OpenSearch cluster.

## Syntax 

```bash
opensearch-benchmark execute_test <arguments>
```

### Common Arguments

See [All Arguments](#all-arguments) for an exhaustive list of arguments

Argument | Description | Required
:--- | :--- |:---
`workload` | The dataset and operations that execute during a test. See [OpenSearch Benchmark Workloads repository](https://github.com/opensearch-project/opensearch-benchmark-workloads) for more details on workloads. | Yes

> `workload-params` | Parameters defined within each workload that can be overwritten, defined in the README of each workload. You can find an example of the eventdata workload [here](https://github.com/opensearch-project/opensearch-benchmark-workloads/tree/main/eventdata#parameters).  | No

Reviewer 3: Nit: "You can find an example of the parameters for the eventdata workload"

> `test_procedure` | Test Procedures define the sequence of operations and parameters for a specific workload. When there is `test_procedure` specified, Benchmark selects the default. Define test procedures within each workload directory. You can fine an example test procedure [here](https://github.com/opensearch-project/opensearch-benchmark-workloads/blob/main/eventdata/test_procedures/default.json). | No

Reviewer 3: 1. Nit: "When there is `test_procedure` specified" --> "When `test_procedure` specified". Also regarding this sentence, I think it should be "When there is no `test_procedure` specified, Benchmark selects the default"? 
3. Spelling error:  **fine**--> **find** in the last sentence
